src/components/UploadDialog/UploadDialog.js

```diff
@@ -9,7 +11,80 @@ import {
   DialogTitle
 } from 'material-ui';
 
+function createImage(pathname) {
+  return {
+    category: null,
+    probability: null,
+    visible: true,
+    identifier: pathname,
+    filename: pathname
+  };
+}
+
+const readFile = (currentFile, that) => {
+  const pathname = currentFile.webkitRelativePath;
+  return e => {
+    const bytes = e.target.result;
+    const image = createImage(pathname);
+    that.imageData.imageDataIndexedDB.push({
+      checksum: pathname,
+      bytes: bytes
+    });
+    that.imageData.imageDataReduxStore.push(image);
```

Comment:
We should add protection against accidental race conditions or inconsistent persists (e.g. IndexedDB.push works but ReduxStore.push doesn't).